reconcile/terraform_vpc_resources/merge_request_manager.py

```diff
@@ -0,0 +1,107 @@
+import logging
```

Comment:
This code is duplicated almost verbatim in at least 3 places that I can find. The code is very generic too which to me makes it a good candidate for a utils library. Do you want to attempt to extract it to a library and instead use the library in integration.py?

We can have jira cards to make other integrations use the utility library, no need to do that as part of this PR

Reply:
That is a great idea.

Reply:
One that notice trying to move this to a utils library is that most of the classes are already a subclass of something commom and the code has some specifics from the VPC merge.
I'm not that confident that there is value in abstracting this further.

`MrData` is singular to this integration and it what is used in `create_merge_request` to check for the existence of an MR.

The shared part is `house_keeping` and the constructor for the `MergeRequestManagerBase` class